plugin.video.embycon/resources/lib/functions.py

```diff
@@ -936,27 +988,61 @@ def searchResults(params):
-        Ids = ",".join(id_list)
-        details_url = ('{server}/emby/Users/{userid}/items' +
-                       '?Ids=' + Ids +
-                       '&Fields={field_filters}' +
-                       '&format=json')
-        details_result = dataManager.GetContent(details_url)
-        log.debug("Search Results Details: {0}", details_result)
-
-        dir_items, detected_type = processDirectory(details_result, progress, params)
-        if dir_items is not None:
-            xbmcplugin.addDirectoryItems(handle, dir_items)
-            xbmcplugin.endOfDirectory(handle, cacheToDisc=False)
-    elif not query_string:
-        xbmcgui.Dialog().ok("No Matches", "No items match your search.")
+    # what type of search was it
+    if item_type == "person":
+        log.debug("Item Search Result")
+        server = downloadUtils.getServer()
+        list_items = []
+        for item in search_hints:
+            person_id = item.get('ItemId')
+            person_name = item.get('Name')
+            image_tag = item.get('PrimaryImageTag')
+            person_thumbnail = downloadUtils.imageUrl(person_id, "Primary", 0, 400, 400, image_tag, server=server)
+
+            action_url = sys.argv[0] + "?mode=NEW_SEARCH_PERSON&person_id=" + person_id
+
+            list_item = xbmcgui.ListItem(label=person_name)
+            list_item.setProperty("id", person_id)
+            if person_thumbnail:
+                art_links = {}
+                art_links["thumb"] = person_thumbnail
+                art_links["poster"] = person_thumbnail
+                list_item.setArt(art_links)
+
+            item_tupple = (action_url, list_item, True)
+            list_items.append(item_tupple)
+
+        xbmcplugin.setContent(handle, 'artists')
+        xbmcplugin.addDirectoryItems(handle, list_items)
+        xbmcplugin.endOfDirectory(handle, cacheToDisc=False)
+
+    else:
+        # extract IDs for details query
+        log.debug("Item Search Result")
+        id_list = []
+        for item in search_hints:
+            item_id = item.get('ItemId')
+            id_list.append(item_id)
+
+        if len(id_list) > 0:
+            Ids = ",".join(id_list)
+            details_url = ('{server}/emby/Users/{userid}/items' +
+                           '?Ids=' + Ids +
+                           '&Fields={field_filters}' +
+                           '&format=json')
+            details_result = dataManager.GetContent(details_url)
+            log.debug("Search Results Details: {0}", details_result)
+
+            # set content type
+            xbmcplugin.setContent(handle, content_type)
+
+            dir_items, detected_type = processDirectory(details_result, progress, params)
+            if dir_items is not None:
+                xbmcplugin.addDirectoryItems(handle, dir_items)
+                xbmcplugin.endOfDirectory(handle, cacheToDisc=False)
+
+        elif not query_string:
+            xbmcgui.Dialog().ok("No Matches", "No items match your search.")
```

Comment:
minor: can you move this to localized strings?

Reply:
will do